src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -304,10 +350,30 @@ impl DeviceManager {
         device_id: String,
     ) -> Result<ArcMutexDevice> {
         let mut block_config = config.clone();
-        block_config.driver_option = self.shared_info.block_driver.clone();
+        let mut is_pmem = false;
+
+        match block_config.driver_option.as_str() {
+            // convert the block driver to kata type
+            VIRTIO_BLOCK_MMIO => {
+                block_config.driver_option = KATA_MMIO_BLK_DEV_TYPE.to_string();
+            }
+            VIRTIO_BLOCK_PCI => {
+                block_config.driver_option = KATA_BLK_DEV_TYPE.to_string();
+            }
+            VIRTIO_PMEM => {
+                block_config.driver_option = KATA_NVDIMM_DEV_TYPE.to_string();
+                is_pmem = true;
+            }
+            _ => {
+                return Err(anyhow!(
+                    "invliad driver type {}",
```

Comment:
"invalid driver type" -> "unsupported driver type" is better.

Reply:
fixed